Marlin/Marlin_main.cpp

```diff
@@ -9946,42 +9946,23 @@ void quickstop_stepper() {
 #if HAS_M206_COMMAND
 
   /**
-   * M428: Set home_offset based on the distance between the
-   *       current_position and the nearest "reference point."
-   *       If an axis is past center its endstop position
-   *       is the reference-point. Otherwise it uses 0. This allows
-   *       the Z offset to be set near the bed when using a max endstop.
-   *
-   *       M428 can't be used more than 2cm away from 0 or an endstop.
+   * M428: Set a persistent offset to the native home position and coordinate
+   *       space by assigning the current position as the native home position.
    *
    *       Use M206 to set these values directly.
    */
   inline void gcode_M428() {
-    bool err = false;
-    LOOP_XYZ(i) {
-      if (axis_homed[i]) {
-        const float base = (current_position[i] > (soft_endstop_min[i] + soft_endstop_max[i]) * 0.5) ? base_home_pos((AxisEnum)i) : 0,
```

Comment:
Recent issues with `M428` are probably primarily related to the broken `report_current_position` function, but might also be related to the fact that we changed soft endstops to use the borders of the printable (bed) region and no longer the hard endstops `(MIN|MAX)_[XYZ]_POS`. To revert to the previous (non-broken) behavior, the `base_min_pos()` and `base_max_pos()` might be better than `soft_endstop_(min|max)` here.

Reply:
The website describes `M428` this way:

> The current position is set to the native home position.

It does fail to describe the case where the home position is something like `X200 Y0 Z0` so that needs to be explained. But the idea is, shortly after you home, you would move the axes a little until you have your preferred position within the bed, but relatively close to the endstop. The `M428` command makes the current point correspond to the configured endstop position by setting `home_offset` appropriately.